DesiMock

```diff
@@ -0,0 +1,195 @@
+
+
+    def get_lyb_index(self, peakix):
+        """
+        Returns the index location of the Ly-B absorption for a given peak index value
+        :param peakix:
+        :return: index location of Ly-B
+        """
+        spectrum_higher = 10**self.loglam[peakix]
+        spectrum_lambda_lower = spectrum_higher * (1025.722 / 1215.67)
+        log_lambda_lower = np.log10(spectrum_lambda_lower)
+        ix_lambda_lower = (np.abs(self.loglam - log_lambda_lower)).argmin()
+        return ix_lambda_lower
+
+    def process(self, model_path):
+        """  The following should follow the algorithm in process_catalog
+        :param model_path:
+        :return:
+        """
+        import json
+        from dla_cnn.data_loader import scan_flux_sample
+        from dla_cnn.localize_model import predictions_ann as predictions_ann_c2
+        from dla_cnn.data_loader import compute_peaks, get_lam_data
+        #from dla_cnn.data_loader import add_abs_to_sightline
+        from dla_cnn.absorption import add_abs_to_sightline
+        from dla_cnn.data_model.Prediction import Prediction
+        # Fluxes
+        fluxes = scan_flux_sample(self.flux, self.loglam, self.z_qso, -1, stride=1)[0]
+        # Model
+        with open(model_path+"_hyperparams.json",'r') as f:
+            hyperparameters = json.load(f)
+        loc_pred, loc_conf, offsets, density_data_flat = predictions_ann_c2(hyperparameters, fluxes, model_path)
+        self.prediction = Prediction(loc_pred=loc_pred, loc_conf=loc_conf, offsets=offsets, density_data=density_data_flat)
+        # Peaks
+        _ = compute_peaks(self)
+        # Absorbers?
+        add_abs_to_sightline(self)
+
+
+class DesiMock:
+    """
+    a class to load all spectrums from a Desi fits file.
+    :attribute wavelength array-like, save the wavelength of all spectrum (all spectrum have same wavelength array)
+    :attribute data, dict, using each spectra's id as its key and a dict of all data we need of this spectra as its value
+      its format like spectra_id: {'FLUX':flux, 'ERROR':error, 'z_qso':z_qso, 'RA':ra, 'DEC':dec, 'DLAS':dlas'information}
+      I think this format may be more readable and practical than simply take the data of each spectra apart.
+    """
+
+    def _init_(self, wavelength = None, data = {}):
+        self.wavelength = wavelength
+        self.data = data
+
+
+    def read_fits_file(self, spec_path, truth_path, zbest_path):
+        """
+        read Desi Mock spectrum from a fits file, load all spectrum as a DesiMock object
+        :param spec_path: spectrum file path
+        :param truth_path: truth file path
+        :param zbest_path: zbest file path
+        :return: self.wavelength,self.data(contained all information we need)
+        """
+        spec = fits.open(spec_path)
+        truth = fits.open(truth_path)
+        zbest = fits.open(zbest_path)
+
+        self.wavelength = np.hstack((spec[2].data.copy(), spec[7].data.copy(), spec[12].data.copy()))
```

Comment:
Add a comment here that says why you are stacking 2, 7, and 12.

Reply:
[2] is blue channel wavelength, [7] is red channel wavelength, and [12] is z channel. I should ask students to do better code comment next time!

Reply:
> Add a comment here that says why you are stacking 2, 7, and 12.

[2] is blue channel wavelength, [7] is red channel wavelength, and [12] is z channel. I should ask students to do better code comment next time!